hnn_core/network_models.py

```diff
@@ -0,0 +1,263 @@
+
+    # xx -> layer5 Basket
+    src_cell = 'L5_basket'
+    target_cell = 'L5_basket'
+    lamtha = 20.
+    loc = 'soma'
+    receptor = 'gabaa'
+    key = f'gbar_L5Basket_{_short_name(target_cell)}'
+    weight = net._params[key]
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha,
+        allow_autapses=False)
+
+    src_cell = 'L5_pyramidal'
+    lamtha = 3.
+    key = f'gbar_L5Pyr_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    src_cell = 'L2_pyramidal'
+    key = f'gbar_L2Pyr_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    return net
+
+
+def law_model(params=None, add_drives_from_params=False):
+    """Instantiate the beta modulated ERP network model.
+
+    Parameters
+    ----------
+    params : dict | None
+        The parameters to use for constructing the network.
+        If None, parameters loaded from beta_erp.json
+        Default: None
+    add_drives_from_params : bool
+        If True, add drives as defined in the params-dict. NB this is mainly
+        for backward-compatibility with HNN GUI, and will be deprecated in a
+        future release. Default: False
+
+    Returns
+    -------
+    net : Instance of Network object
+        Network object used to store the default network.
+        All connections defining the default network will be
+        appeneded to net.connectivity.
+    Notes
+    -----
+    Model reproduces results from Law et al. 2021
+    This model differs from the default network model in several
+    parameters including: (fill in later)
+    """
+
+    hnn_core_root = op.dirname(hnn_core.__file__)
+    params_fname = op.join(hnn_core_root, 'param', 'default.json')
+    if params is None:
+        params = read_params(params_fname)
+
+    params['tstop'] = 300.0
+    net = jones_2009_model(params)
+
+    # Update biophysics (increase gabab duration of inhibition)
+    net.cell_types['L2_pyramidal'].p_syn['gabab']['tau1'] = 45.0
+    net.cell_types['L2_pyramidal'].p_syn['gabab']['tau2'] = 200.0
+    net.cell_types['L5_pyramidal'].p_syn['gabab']['tau1'] = 45.0
+    net.cell_types['L5_pyramidal'].p_syn['gabab']['tau2'] = 200.0
+
+    # Decrease L5_pyramidal -> L5_pyramidal nmda weight
+    net.connectivity[2]['nc_dict']['A_weight'] = 0.0004
+
+    # Modify L5_basket -> L5_pyramidal inhibition
+    net.connectivity[6]['nc_dict']['A_weight'] = 0.02  # gabaa
+    net.connectivity[7]['nc_dict']['A_weight'] = 0.005  # gabab
+
+    # Remove L5 pyramidal somatic and basal dendrite calcium channels
+    for sec in ['soma', 'basal_1', 'basal_2', 'basal_3']:
+        net.cell_types['L5_pyramidal'].p_secs[
+            sec]['mechs']['ca']['gbar_ca'] = 0.0
```

Comment:
I think it might be better to do:

```suggestion
        del net.cell_types['L5_pyramidal'].p_secs[
            sec]['mechs']['ca']
```